src/systemctl/systemctl.c

```diff
@@ -440,7 +441,8 @@ static int systemctl_parse_argv(int argc, char *argv[]) {
                 { "global",              no_argument,       NULL, ARG_GLOBAL              },
                 { "wait",                no_argument,       NULL, ARG_WAIT                },
                 { "no-block",            no_argument,       NULL, ARG_NO_BLOCK            },
-                { "no-legend",           no_argument,       NULL, ARG_NO_LEGEND           },
+                { "legend",              optional_argument, NULL, ARG_LEGEND              },
```

Comment:
please don't use `optional_argument`, it's seriously flawed. the thing is that `systemctl --legend yes` is suddenly ambigous. It can mean `systemctl --legend=yes list-units` or `systemctl --legend=yes yes`.

We really should never use `optional_argument`. It was a mistake starting with that ever, but going forward we should not add new ones.

The only place where one can get away with `optional_argument` is in places where we have no positional arguments at all..

Also, there's no reason to support `--legend` without argument, given that the legend is the default anyway...

Hence, I'd really just do `--legend` with `required_argument` + `--no-legend` (for compat), but stay away from `--legend` with `no_argument` and from `--legend` with `optional_argument`

Reply:
Updated with the suggested changes.